src/tr2/game/savegame/common.c

```diff
@@ -42,25 +194,441 @@ int32_t Savegame_GetSlotCount(void)
+            Inv_RequestItem(O_SHOTGUN_AMMO_ITEM) * SHOTGUN_AMMO_QTY;
+    }
+
+    if (Inv_RequestItem(O_MAGNUM_ITEM)) {
+        resume->has_magnums = 1;
+        resume->magnum_ammo = g_Lara.magnum_ammo.ammo;
+    } else {
+        resume->has_magnums = 0;
+        resume->magnum_ammo =
+            Inv_RequestItem(O_MAGNUM_AMMO_ITEM) * MAGNUM_AMMO_QTY;
+    }
+
+    if (Inv_RequestItem(O_UZI_ITEM)) {
+        resume->has_uzis = 1;
+        resume->uzi_ammo = g_Lara.uzi_ammo.ammo;
+    } else {
+        resume->has_uzis = 0;
+        resume->uzi_ammo = Inv_RequestItem(O_UZI_AMMO_ITEM) * UZI_AMMO_QTY;
+    }
+
+    if (Inv_RequestItem(O_M16_ITEM)) {
+        resume->has_m16 = 1;
+        resume->m16_ammo = g_Lara.m16_ammo.ammo;
+    } else {
+        resume->has_m16 = 0;
+        resume->m16_ammo = Inv_RequestItem(O_M16_AMMO_ITEM) * M16_AMMO_QTY;
+    }
+
+    if (Inv_RequestItem(O_HARPOON_ITEM)) {
+        resume->has_harpoon = 1;
+        resume->harpoon_ammo = g_Lara.harpoon_ammo.ammo;
+    } else {
+        resume->has_harpoon = 0;
+        resume->harpoon_ammo =
+            Inv_RequestItem(O_HARPOON_AMMO_ITEM) * HARPOON_AMMO_QTY;
+    }
+
+    if (Inv_RequestItem(O_GRENADE_ITEM)) {
+        resume->has_grenade = 1;
+        resume->grenade_ammo = g_Lara.grenade_ammo.ammo;
+    } else {
+        resume->has_grenade = 0;
+        resume->grenade_ammo =
+            Inv_RequestItem(O_GRENADE_AMMO_ITEM) * GRENADE_AMMO_QTY;
+    }
+
+    resume->flares = Inv_RequestItem(O_FLARE_ITEM);
+    resume->small_medipacks = Inv_RequestItem(O_SMALL_MEDIPACK_ITEM);
+    resume->large_medipacks = Inv_RequestItem(O_LARGE_MEDIPACK_ITEM);
+
+    if (g_Lara.gun_type == LGT_FLARE) {
+        resume->gun_type = g_Lara.last_gun_type;
+    } else {
+        resume->gun_type = g_Lara.gun_type;
+    }
+    resume->gun_status = LGS_ARMLESS;
 }
 
 bool Savegame_Save(const int32_t slot_idx)
 {
-    CreateSaveGameInfo();
-    S_SaveGame(slot_idx);
-    GetSavedGamesList(&g_LoadGameRequester);
-    return true;
+    bool result = false;
+    Savegame_BindSlot(slot_idx);
+
+    File_CreateDirectory(SAVES_DIR);
+
+    const GF_LEVEL *const current_level = Game_GetCurrentLevel();
+    const char *const level_title = current_level->title;
+
+    Savegame_PersistGameToCurrentInfo(current_level);
+
+    SAVEGAME_INFO *const savegame_info = &m_SavegameInfo[slot_idx];
+    const bool was_slot_empty = savegame_info->full_path == nullptr;
+
+    for (int32_t i = 0; i < m_StrategyCount; i++) {
+        const SAVEGAME_STRATEGY strategy = m_Strategies[i];
+        if (!strategy.allow_save || strategy.save_to_file_func == nullptr) {
+            continue;
+        }
+
+        char *file_name =
+            String_Format(strategy.get_save_file_pattern_func(), slot_idx);
+        char *full_path = String_Format("%s/%s", SAVES_DIR, file_name);
+        MYFILE *const fp = File_Open(full_path, FILE_OPEN_WRITE);
+        if (fp != nullptr) {
+            m_SaveCounter++;
```

Comment:
Let's move this to the if outside the for loop as theoretically it can increment this counter once per each successful saving strategy.

Reply:
Done, I've had to add a decrement check at the end too if the result isn't true - the idea is the strategies access it with `Savegame_GetCounter` so we need to have incremented it beforehand.